tests/Test/Metrics/TypeClassComplexitySpec.hs

```diff
@@ -0,0 +1,30 @@
+module Test.Metrics.TypeClassComplexitySpec where
+
+
+import Test.Hspec
+import Test.Utilities
+
+import Language.Haskell.Exts.Extension
+import Language.Haskell.Homplexity.TypeClassComplexity
+import Language.Haskell.Homplexity.CodeFragment (TypeClass, occurs)
+import Language.Haskell.Homplexity.Metric (measure)
+import Language.Haskell.Homplexity.Parse (parseSource)
+
+spec :: Spec
+spec = describe "tests for type class complexity" $ do
+  it "correctly counts methods and values" $ flip shouldReturn (5 :: NonTypeDeclCount) $ do
+    result <- parseSource [] (testFile "TypeClassComplexityTest.hs")
+    p <- case result of
+      Left  logs      -> error $ show logs
+      Right (prog, _) -> pure prog
+    let tyClassDecls = occurs p :: [TypeClass]
+    pure $ sum $ map measure tyClassDecls
+
+  it "correctly counts associated types" $ flip shouldReturn (5 :: AssocTypeCount) $ do
+    result <- parseSource [EnableExtension TypeFamilies] (testFile "TypeClassComplexityTest.hs")
```

Comment:
We might want to extract a common `parseSrcAndComputeMetric` function for both.

Reply:
I think it is too early to generalize this functionality, but as it turns out, it is also a nice opportunity to showcase a possible improvement to the codebase. 

I think `Proxy` values can be subsumed by using `AllowAmbigousTypes` and `TypeApplications`. Commit 51b79ec demonstrates this.